File: task13/main.py
# Вариант с файлом:
# with open("input.txt") as f:
#     lines = f.read().splitlines()
#     arr = list(map(int, lines[0].split()))
#     target = int(lines[1])

def binary_search(arr, target):
    left, right = 0, len(arr) - 1

    while left <= right:
        mid = (left + right) // 2
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1

    return -1

arr = list(map(int, input("Введите отсортированный массив через пробел: ").split()))
target = int(input("Введите искомый элемент: "))
print(binary_search(arr, target))

File: task13/main.cpp
#include <iostream>
#include <vector>
// Вариант с файлом:
// #include <fstream>
using namespace std;

int binarySearch(vector<int>& arr, int target) {
    int left = 0, right = arr.size() - 1;

    while (left <= right) {
        int mid = left + (right - left) / 2;
        if (arr[mid] == target)
            return mid;
        else if (arr[mid] < target)
            left = mid + 1;
        else
            right = mid - 1;
    }

    return -1;
}

int main() {
    int n;
    cout << "Введите количество элементов (отсортированный массив): ";
    cin >> n;

    // Вариант с файлом:
    // ifstream f("input.txt");
    // f >> n;
    // vector<int> arr(n);
    // for (int& x : arr) f >> x;
    // int target; f >> target;

    vector<int> arr(n);
    cout << "Введите элементы: ";
    for (int& x : arr) cin >> x;

    int target;
    cout << "Введите искомый элемент: ";
    cin >> target;

    cout << binarySearch(arr, target) << endl;
    return 0;
}